src/main/java/de/medizininformatik_initiative/process/data_transfer/service/InsertData.java

```diff
@@ -116,6 +121,48 @@ private List<IdType> storeData(Variables variables, FhirClient fhirClient, Bundl
 		return idsOfCreatedResources;
 	}
 
+	private Bundle checkAndAdaptBundleForExistingData(FhirClient fhirClient, Bundle bundle, String sendingOrganization,
+			String projectIdentifier, Task task)
+	{
+		Bundle searchResult = fhirClient.getGenericFhirClient().search().forResource(DocumentReference.class)
+				.where(DocumentReference.IDENTIFIER.exactly()
+						.systemAndCode(ConstantsBase.NAMINGSYSTEM_MII_PROJECT_IDENTIFIER, projectIdentifier))
+				.and(DocumentReference.AUTHOR.hasChainedProperty(Organization.IDENTIFIER.exactly()
+						.systemAndCode(NamingSystems.OrganizationIdentifier.SID, sendingOrganization)))
+				.returnBundle(Bundle.class).execute();
+
+		List<DocumentReference> existingDocumentReferences = searchResult.getEntry().stream()
+				.filter(Bundle.BundleEntryComponent::hasResource).map(Bundle.BundleEntryComponent::getResource)
+				.filter(r -> r instanceof DocumentReference).map(r -> (DocumentReference) r).toList();
+
+		if (existingDocumentReferences.size() < 1)
+			return bundle;
+
+		if (existingDocumentReferences.size() > 1)
+			logger.warn(
+					"Found more than one DocumentReference for project-identifier '{}' authored by '{}', using the first",
+					projectIdentifier, sendingOrganization);
+
+		DocumentReference existingDocumentReference = existingDocumentReferences.get(0);
+		String existingDocumentReferenceId = existingDocumentReference.getIdElement().getIdPart();
+
+		logger.info(
+				"DocumentReference for project-identifier '{}' authored by '{}' already exists, updating data-set on FHIR server with baseUrl '{}' in Task with id '{}'",
+				projectIdentifier, sendingOrganization, fhirClient.getFhirBaseUrl(), task.getId());
+
+		bundle.getEntry().stream().filter(Bundle.BundleEntryComponent::hasResource)
+				.filter(e -> e.getResource() instanceof DocumentReference)
+				.filter(Bundle.BundleEntryComponent::hasRequest).filter(Bundle.BundleEntryComponent::hasResource)
+				.forEach(e ->
+				{
+					e.getRequest().setMethod(Bundle.HTTPVerb.PUT)
+							.setUrl(ResourceType.DocumentReference.name() + "/" + existingDocumentReferenceId);
+					e.getResource().setId(existingDocumentReferenceId);
+				});
+
+		return bundle;
+	}
```

Comment:
Consider moving to `mii-processes-common` module.

Reply:
Good idea, with implementation of the DSF API v2 for process development, we will also redesign the `mii-processes-common` repository. I will move this code as part of the redesign.